agent/workers/envelope/spans_envelope.go

```diff
@@ -0,0 +1,42 @@
+package envelope
+
+import (
+	agentProto "github.com/kubeshop/tracetest/agent/proto"
+	"google.golang.org/protobuf/proto"
+)
+
+// EnvelopeSpans get a list of spans and batch them into a packet that does not
+// surpasses the maxPacketSize restriction. When maxPacketSize is reached, no
+// more spans are added to the packet,
+func EnvelopeSpans(spans []*agentProto.Span, maxPacketSize int) []*agentProto.Span {
+	envelope := make([]*agentProto.Span, 0, len(spans))
+	currentSize := 0
+
+	// There's a weird scenario that must be covered here: imagine a span so big it is bigger than maxPacketSize.
+	// It is impossible to send a span like this, so in this case, we classify those spans as "large spans" and we allow
+	// `largeSpansPerPacket` per packet.
```

Comment:
This is a great point, I believe there is an issue around this very same subject in one of the otel collector repos, I'm glad you thought about it too!